service/src/main/java/greencity/service/NotificationProducerServiceImpl.java

```diff
@@ -0,0 +1,187 @@
+     * Sends a notification about a comment on an article.
+     *
+     * @param articleId       ID of the article
+     * @param articleTitle    Title of the article
+     * @param authorId        ID of the article author (who will receive the
+     *                        notification)
+     * @param commentatorId   ID of the user who commented
+     * @param commentatorName Name of the user who commented
+     */
+    public void sendCommentNotification(Long articleId, String articleTitle,
+        Long authorId, Long commentatorId, String commentatorName) {
+        sendNotification(
+            NotificationType.COMMENT_CREATED,
+            authorId,
+            commentatorId,
+            commentatorName,
+            articleId,
+            articleTitle,
+            "ARTICLE");
+    }
+
+    /**
+     * Sends a notification about a comment on an article or other object.
+     *
+     * @param articleId       ID of the article or object where the comment was made
+     * @param articleTitle    Title of the article or object where the comment was made
+     * @param authorId        ID of the article or object author (who will receive the notification)
+     * @param commentatorId   ID of the user who made the comment
+     * @param commentatorName Name of the user who made the comment
+     * @param objectType      Type of the object (e.g., "ARTICLE", "EVENT") where the comment was made
+     */
+    public void sendCommentNotification(Long articleId, String articleTitle,
+                                        Long authorId, Long commentatorId, String commentatorName, String objectType) {
+        sendNotification(
+                NotificationType.COMMENT_CREATED,
+                authorId,
+                commentatorId,
+                commentatorName,
+                articleId,
+                articleTitle,
+                objectType);
+    }
+
+    /**
+     * Sends a notification about a like on an article.
+     *
+     * @param articleId    ID of the article
+     * @param articleTitle Title of the article
+     * @param authorId     ID of the article author (who will receive the
+     *                     notification)
+     * @param likerId      ID of the user who liked
+     * @param likerName    Name of the user who liked
+     */
+    public void sendLikeNotification(Long articleId, String articleTitle,
+        Long authorId, Long likerId, String likerName) {
+        sendNotification(
+            NotificationType.ARTICLE_LIKED,
+            authorId,
+            likerId,
+            likerName,
+            articleId,
+            articleTitle,
+            "ARTICLE");
+    }
+
+
+    /**
+     * Sends a notification about a like on an article or other*/
+    public void sendLikeNotification(Long articleId, String articleTitle,
+                                     Long authorId, Long likerId, String likerName, String objectType) {
+        sendNotification(
+                NotificationType.ARTICLE_LIKED,
```

Comment:
For habits, you need to create others NotificationType, like HABIT_LIKED
```suggestion
                NotificationType.ARTICLE_LIKED,
```